### PR TITLE
add oh 529 plan deduction

### LINKED ISSUE #3266: Ohio 529 plan deduction
Tax form: https://tax.ohio.gov/static/forms/ohio_individual/individual/2022/it1040-sd100-instruction-booklet.pdf#page=18

Legal code: https://codes.ohio.gov/ohio-revised-code/section-5747.01

### PR DESCRIPTION
**Pull request by Linfeng-ll**
Fixes #3266 

github:all

### CODE REVIEW

File: policyengine_us/variables/gov/states/oh/tax/income/deductions/oh_529_plan_deduction.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    definition_period = YEAR
+    reference = (
+        "https://tax.ohio.gov/static/forms/ohio_individual/individual/2021/pit-it1040-booklet.pdf#page=18",
+    )
+    defined_for = StateCode.OH
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.oh.tax.income.deductions
+        contribution_529 = tax_unit("investment_in_529_plan", period)
+        deduction_529 = min_(contribution_529, p.plan_529_deduction_threshold)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/oh/tax/income/deductions/oh_529_plan_deduction.py:18]:
```suggestion
        capped_amount = min_(contribution_529, p.plan_529_deduction_threshold)
```

File: policyengine_us/variables/gov/states/oh/tax/income/deductions/oh_529_plan_deduction.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        "https://tax.ohio.gov/static/forms/ohio_individual/individual/2021/pit-it1040-booklet.pdf#page=18",
+    )
+    defined_for = StateCode.OH
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.oh.tax.income.deductions
+        contribution_529 = tax_unit("investment_in_529_plan", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/oh/tax/income/deductions/oh_529_plan_deduction.py:17]:
```suggestion
        contribution_amount = tax_unit("investment_in_529_plan", period)
```

File: policyengine_us/variables/gov/states/oh/tax/income/deductions/oh_529_plan_deduction.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+from policyengine_us.model_api import *
+
+
+class oh_529_plan_deduction(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Contributions to an Ohio 529 plan deduction"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/oh/tax/income/deductions/oh_529_plan_deduction.py:7]:
```suggestion
    label = "Ohio contributions to 529 plan deduction"
```

File: policyengine_us/parameters/gov/states/oh/tax/income/deductions/plan_529_deduction_threshold.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/oh/tax/income/deductions/plan_529_deduction_threshold.yaml:1]: make a folder called `529_plan` and then a file called `max_amount`

File: policyengine_us/parameters/gov/states/oh/tax/income/deductions/plan_529_deduction_threshold.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+description: Ohio limits its contributions to an Ohio 529 plan deduction below this amount.
+values:
+  2021-01-01: 4_000
+metadata:
+  unit: currency-USD
+  period: year
+  label: Ohio 529 plan deduction threshold
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/oh/tax/income/deductions/plan_529_deduction_threshold.yaml:7]:
```suggestion
  label: Ohio 529 plan contribution deduction max amount
```

File: policyengine_us/parameters/gov/states/oh/tax/income/deductions/plan_529_deduction_threshold.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/oh/tax/income/deductions/plan_529_deduction_threshold.yaml:1]: make a deductions.yaml file and include the oh_529_plan_deduction.yaml as first deduction

File: policyengine_us/variables/gov/states/oh/tax/income/deductions/oh_529_plan_deduction.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    )
+    defined_for = StateCode.OH
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.oh.tax.income.deductions.plan_529_contributions
+        contribution_amount = tax_unit("investment_in_529_plan", period)
+        capped_amount = min_(contribution_amount, p.max_amount)
+        return max_(capped_amount, 0)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/oh/tax/income/deductions/oh_529_plan_deduction.py:21]:
```suggestion
```

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/parameters/gov/states/oh/tax/income/deductions/plan_529_contributions/max_amount.yaml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+description: Ohio limits its contributions to an Ohio 529 plan deduction below this amount.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/oh/tax/income/deductions/plan_529_contributions/max_amount.yaml:1]:
```suggestion
description: Ohio deducts 529 plan contributions up to this amount from taxable income.
```

File: policyengine_us/parameters/gov/states/oh/tax/income/deductions/plan_529_contributions/max_amount.yaml [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+values:
+  2021-01-01: 4_000
+metadata:
+  unit: currency-USD
+  period: year
+  label: Ohio 529 plan contribution deduction max amount
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/oh/tax/income/deductions/plan_529_contributions/max_amount.yaml:7]: rename cap.yaml
```suggestion
  label: Ohio 529 plan contribution deduction cap
```

File: changelog_entry.yaml [hunk 8]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Ohio 295 plan deduction.
```
Comment (MaxGhenis) [inline changelog_entry.yaml:4]:
```suggestion
    - Ohio 529 plan deduction.
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Please also tie this to an issue and include github:all in the description.

For future reference, we don't need to model programs like these at this time, since they're not part of [taxsim35](https://taxsim.nber.org/taxsim35/). Filing issues before PRs gives a chance to discuss the importance of the programs.

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: could you run a `make install` and then a `git pull upstream master` to see if the conflicts will be fixed

File: policyengine_us/variables/gov/states/oh/tax/income/deductions/oh_529_plan_deduction.py [hunk 9]
```diff
@@ -0,0 +1,20 @@
+from policyengine_us.model_api import *
+
+
+class oh_529_plan_deduction(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Ohio contributions to 529 plan deduction"
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/oh/tax/income/deductions/oh_529_plan_deduction.py:7]:
```suggestion
    label = "Ohio deduction for contributions to 529 plans"
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**